src/tests/domParsers/khanAcademyVideo_tests.ts

```diff
@@ -0,0 +1,102 @@
+/// <reference path="../../../typings/main/ambient/qunit/qunit.d.ts" />
+
+import {VideoUtils} from "../../scripts/domParsers/videoUtils";
+
+let supportedKhanAcademyUrls = [
+	"https://www.khanacademy.org/humanities/art-1010/beginners-guide-20-21/v/representation-abstraction-looking-at-millais-and-newman",
+	"https://www.khanacademy.org/computing/computer-programming/programming/intro-to-programming/v/programming-intro",
+	"https://www.khanacademy.org/college-admissions/applying-to-college/introduction-atc/v/sal-khans-story-applying-to-college"
+];
+
+let unsupportedKhanAcademyUrls = [
+	"https://www.khanacademy.org",
+	"https://www.khanacademy.org/college-admissions/applying-to-college/college-application-process/a/filling-out-the-college-application-common-application-walkthrough",
+	"https://www.khanacademy.org/college-admissions/applying-to-college?ref=our_favorite_topics"
+];
+
+let pageContentHtmlWrapperPrepend = "<html lang='en'> <body>";
+let pageContentHtmlWrapperAppend = "</body> </html>";
+
+let pageContentWithNoClipId = [
+	undefined,
+	"",
+	"<div id='clip'> </div>", // no clip id
+	"<div id='clip_abcdefg'> </div>" // invalid clip id
+];
+
+let pageContentWithHyphenatedVideoIds = [
+	"<div id='video_8-5DTsl1V5k'> </div>", // standard
+	"<div id = ' video_8-5DTsl1V5k ' > </div> ", // weird whitespace
+	"<DIV ID='Video_8-5DTsl1V5k'> </DIV>", // weird casing
+	"<div id=\"video_8-5DTsl1V5k\"> </div>", // double quotations,
+];
+
+let pageContentWithDataYoutubeIds = [
+	"<div data-youtubeid='video_8-5DTsl1V5k'> </div>", // standard
+	"<div data-youtubeid = ' video_8-5DTsl1V5k ' > </div> ", // weird whitespace
+	"<DIV data-youtubeid='Video_8-5DTsl1V5k'> </DIV>", // weird casing
+	"<div data-youtubeid=\"video_8-5DTsl1V5k\"> </div>" // double quotations
+];
+
+let pageContentWithMultipleHyphenatedVideoIds = [
+	"<div id='video_8-5DTsl1V5k'></div> <DIV ID='video_8-53sl1V5k'></DIV> <div id='video_4ba196611'> </div>",
+];
+
+QUnit.module("videoUtils");
+
+test("videoDomainIfSupported", () => {
```

Comment:
The test name should be fleshed out so that if it fails, the person reading the console will know immediately what is going on